samples/snippets/storage_trace_quickstart.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# Copyright 2024 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+"""
+Sample that exports OpenTelemetry Traces collected from the Storage client to Cloud Trace.
+"""
+
+
+def run_quickstart(bucket_name, blob_name, contents):
+    # [START storage_enable_otel_tracing]
+
+    from opentelemetry import trace
+    from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+    from opentelemetry.instrumentation.requests import RequestsInstrumentor
+    from opentelemetry.resourcedetector.gcp_resource_detector import (
+        GoogleCloudResourceDetector,
+    )
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor
+    from opentelemetry.sdk.trace.sampling import TraceIdRatioBased
+
+    from google.cloud import storage
+
+    # The ID of your GCS bucket
+    # bucket_name = "your-bucket-name"
+    # The ID of your GCS object
+    # blob_name = "your-object-name"
+    # The contents to upload to the file
+    # contents = "The quick brown fox jumps over the lazy dog."
+
+    # In this sample, we use Google Cloud Trace to export the OpenTelemetry
+    # traces: https://cloud.google.com/trace/docs/setup/python-ot
+    # Choose and configure the exporter for your set up accordingly.
+
+    # Configure the sampling rate to control trace ingestion volume.
+    # Trace sampling rate must be in the range [0.0,1.0].
+    tracing_rate = 1.0
+    tracer_provider = TracerProvider(
+        sampler=TraceIdRatioBased(tracing_rate),
+        resource=GoogleCloudResourceDetector().detect(),
+    )
+
+    # Export to Google Cloud Trace.
+    tracer_provider.add_span_processor(BatchSpanProcessor(CloudTraceSpanExporter()))
+    trace.set_tracer_provider(tracer_provider)
+
+    # Optional yet recommended: enable traces emitted from the requests HTTP library.
+    RequestsInstrumentor().instrument(tracer_provider=tracer_provider)
+
+    # Instantiate a storage client and perform a write and read workload.
+    storage_client = storage.Client()
```

Comment:
Do you need a parent span that groups the requests being made?

Reply:
Updated to create a root span that groups all the requests, thanks!